### PR TITLE
DO NOT MERGE YET. Release 0.4

### PR DESCRIPTION
**Pull request by mmagician**
## Description

See comments in the `makefile` for the proposed process of doing releases.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Review by Pratyush — APPROVED** [review]

**Review by weikengchen — APPROVED** [review]

**Comment by mmagician** [comment]
I have a slightly pedantic comment regarding this current approach actually: it would be nice if the commit where a tag was created from matched the one on master, e.g. like the releases in [RustCrypto RSA](https://github.com/RustCrypto/RSA). Currently due to a merge commit, this isn´t happening and the tag's commit is on a release branch only.

I would propose a 2-step release, kind of like outlined here: https://github.com/crate-ci/cargo-release/issues/119

With some changes [#54](https://github.com/cargo-bins/release-pr/issues/54) & [#53](https://github.com/cargo-bins/release-pr/issues/53) to release-pr, it should be possible to add an action to the workflows and automate the first step of the release process. The second step should then be easy to automate too, but initially can be a manual run of (an adapted) `make release`.

**Comment by weikengchen** [comment]
OK. The side effect has been fixed.

**Comment by weikengchen** [comment]
(Added the "DO NOT MERGE YET" to prevent incidental merge. I believe that for now this is a tracking PR, and we will merge it to master when everything is finalized.)

**Review by Pratyush — APPROVED** [review]

**Pull request merged**
